tests/system_tests_ssl.py

```diff
@@ -780,5 +781,148 @@ def test_connected_tls_sasl_routers(self):
+            ('sslProfile', {'name': 'ssl-profile-tls-all',
+                            'caCertFile': cls.ssl_file('ca-certificate-BAD.pem'),
+                            'certFile': cls.ssl_file('client-certificate.pem'),
+                            'privateKeyFile': cls.ssl_file('client-private-key.pem'),
+                            'ciphers': 'ECDH+AESGCM:DH+AESGCM:ECDH+AES256:DH+AES256:ECDH+AES128:' \
+                                       'DH+AES:RSA+AESGCM:RSA+AES:!aNULL:!MD5:!DSS',
+                            'password': 'client-password'})
+        ])
+
+        cls.routers.append(cls.tester.qdrouterd("A", config_a, wait=False))
+        cls.routers.append(cls.tester.qdrouterd("B", config_b, wait=False))
+
+        # Wait until A is running
+        cls.routers[0].wait_ports()
+
+        # Can't wait until B is connected because it's not supposed to connect.
+
+    def get_router_nodes(self):
+        """
+        Retrieves connected router nodes from QDR.A
+        :return: list of connected router id's
+        """
+        if not SASL.extended():
+            self.skipTest("Cyrus library not available. skipping test")
+
+        url = Url("amqp://0.0.0.0:%d/$management" % self.PORT_NO_SSL)
+        node = Node.connect(url)
+        response = node.query(type="org.apache.qpid.dispatch.router.node", attribute_names=["id"])
+        router_nodes = []
+        for resp in response.get_dicts():
+            router_nodes.append(resp['id'])
+        node.close()
+        return router_nodes
+
+    @SkipIfNeeded(RouterTestSslBase.DISABLE_SSL_TESTING or not SASL.extended(),
+                  "Cyrus library not available. skipping test")
+    def test_connected_misconfigured_tls_sasl_routers(self):
+        """
+        Prove that improperly configured ssl-enabled connector prevents the router
+        from joining the network
+        """
+        if not SASL.extended():
+            self.skipTest("Cyrus library not available. skipping test")
+
+        # Poll for a while until the connector error shows up in router B's log
+        pattern = " SERVER (error) SSL CA configuration failed"
+        sleep_time = 0.1 # seconds
+        poll_duration = 10.0 # seconds
+        verified = False
+        for tries in range(int(poll_duration / sleep_time)):
+            with  open('../setUpClass/B.log', 'r') as router_log:
+                log_lines = router_log.read().split("\n")
+            e_lines = [s for s in log_lines if pattern in s]
+            verified = len(e_lines) > 0
+            if verified:
+                break;
+            time.sleep(sleep_time)
+        self.assertTrue(verified, "Log line containing '%s' not seen in QDR.B log" % pattern)
+
+        # Show that router A does not have router B in its network
+        router_nodes = self.get_router_nodes()
+        self.assertTrue(router_nodes)
+        node = "QDR.B"
+        self.assertTrue(node not in router_nodes, "%s should not be connected" % node)
```

Comment:
Use assertNotIn instead of assertTrue
https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertNotIn
https://docs.python.org/2/library/unittest.html#unittest.TestCase.assertNotIn

Reply:
fixed